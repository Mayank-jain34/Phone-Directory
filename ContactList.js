import React from 'react';
import ContactListRT from './ContactList.rt';
import {
  browserHistory
}
from 'react-router';
var ContactList = React.createClass({
  getInitialState: function() {
    return {
      contacts: [{
        id: 1,
        name: "A1",
        phoneNo: "1234567890"
      }, {
        id: 2,
        name: "A2",
        phoneNo: "1234567890"
      }, {
        id: 3,
        name: "A3",
        phoneNo: "1234567890"
      }]
    };
  },
  _goToContactDetails: function(id) {
    console.log(id);
    browserHistory.push(`/contact/${id}`);
  },
  render: ContactListRT
});

export default ContactList;
