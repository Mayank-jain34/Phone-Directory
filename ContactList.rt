<rt-import name="ListGroup" from="react-bootstrap/lib/ListGroup" />
<rt-import name="ListGroupItem" from="react-bootstrap/lib/ListGroupItem" />

<ListGroup>
   <ListGroupItem  key={contact.id} onClick="()=>this._goToContactDetails(contact.id)"
      rt-repeat="contact in this.state.contacts" header={contact.name}>
      {contact.phoneNo}
  </ListGroupItem>
</ListGroup>
