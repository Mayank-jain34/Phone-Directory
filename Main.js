import React , {Component} from 'react';
import { Navbar,Grid } from 'react-bootstrap';

class Main extends Component {
  render() {

    return (
      <div>
          <Navbar inverse>
            <Navbar.Header>
              <Navbar.Brand>Phone Directory</Navbar.Brand>
              </Navbar.Header>
          </Navbar>
          <Grid>
            {this.props.children}
          </Grid>
          <Navbar inverse fixedBottom = {true}>
            <Navbar.Header>
            </Navbar.Header>
          </Navbar>
      </div>
    );
  }
}

export default Main;
