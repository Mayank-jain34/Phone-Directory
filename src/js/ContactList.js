import React from 'react';
import ContactListRT from './ContactList.rt';
import {  browserHistory } from 'react-router';
import Store from './store/store';

class ContactList extends React.Component {
  constructor() {
    super();
    this.state =  {
      contacts: []
    };
  }
  componentDidMount() {
    this.setState({contacts : Store.getContacts()});
  }
  _goToContactDetails(id) {
    browserHistory.push(`/contact/${id}`);
  }
  render() {
    return ContactListRT.apply(this);
  }
};

export default ContactList;
