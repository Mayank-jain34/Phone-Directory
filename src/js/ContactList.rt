<rt-import name="ListGroup" from="react-bootstrap/lib/ListGroup" />
<rt-import name="ListGroupItem" from="react-bootstrap/lib/ListGroupItem" />

<ListGroup>
  <rt-virtual rt-repeat="contact in this.state.contacts">
    <ListGroupItem key={contact.id} onClick="()=>this._goToContactDetails(contact.id)"
         header={contact.name}>
        {contact.phoneNo}
        </ListGroupItem>
  </virtual>
</ListGroup>
