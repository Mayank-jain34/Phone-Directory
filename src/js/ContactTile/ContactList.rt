<rt-import name = "Button" from = "react-bootstrap/lib/Button">
