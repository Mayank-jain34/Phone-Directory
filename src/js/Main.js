import React , {Component} from 'react';
import { Navbar,Grid } from 'react-bootstrap';
import {browserHistory} from 'react-router';
class Main extends Component {
  constructor() {
    super();
    this._goToHome = this._goToHome.bind(this);
  }
  _goToHome() {
    browserHistory.push('/');
  }
  render() {
    return (
      <div>
          <Navbar inverse>
            <Navbar.Header>
              <Navbar.Brand onClick={this._goToHome}>Phone Directory</Navbar.Brand>
              </Navbar.Header>
          </Navbar>
          <Grid>
            {this.props.children}
          </Grid>
          <Navbar inverse fixedBottom = {true}>
          </Navbar>
      </div>
    );
  }
}

export default Main;
