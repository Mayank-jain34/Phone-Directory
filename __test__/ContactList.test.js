import React from 'react';
import renderer from 'react-test-renderer';
jest.autoMockOff();
import ContactList from '../src/js/ContactList.js';
jest.autoMockOn();
import Main from '../src/js/Main.js';
jest.mock('react-dom');
describe('Componet Testing',()=>{
  it("checking for Main",()=>{
    const component = renderer.create(
     <Main></Main>
  );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("checking for ContactList",()=>{
    const component = renderer.create(
      <ContactList></ContactList>
  );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
