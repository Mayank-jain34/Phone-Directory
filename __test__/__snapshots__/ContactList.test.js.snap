exports[`Componet Testing checking for ContactList 1`] = `
<div>
  <nav
    className="navbar navbar-inverse">
    <div
      className="container">
      <div
        className="navbar-header">
        <span
          className="navbar-brand"
          onClick={[Function bound _goToHome]}>
          Phone Directory
        </span>
      </div>
    </div>
  </nav>
  <div
    className="container" />
  <nav
    className="navbar navbar-inverse navbar-fixed-bottom">
    <div
      className="container" />
  </nav>
</div>
`;
